src/main/java/io/nats/client/impl/MessageQueue.java

```diff
@@ -37,12 +37,17 @@ class MessageQueue {
     protected final LinkedBlockingQueue<NatsMessage> queue;
     protected final Lock filterLock;
     protected final boolean discardWhenFull;
+    protected final long offerTimeoutMillis;
 
     // Poison pill is a graphic, but common term for an item that breaks loops or stop something.
     // In this class the poisonPill is used to break out of timed waits on the blocking queue.
     // A simple == is used to check if any message in the queue is this message.
     protected final NatsMessage poisonPill;
 
+    MessageQueue(boolean singleReaderMode, Duration requestCleanupInterval) {
+        this(singleReaderMode, -1, false, requestCleanupInterval);
+    }
+

```

Comment:
I combined constructors which required cleanup up the places that the object is constructed. This is better because the calls are now more explicit.